### PR TITLE
CheckDockerExists

### PR DESCRIPTION
**Pull request by gablg1**
Checks if Docker exists before attempting to install it

### CODE REVIEW

File: perfkitbenchmarker/linux_virtual_machine.py [hunk 1]
```diff
@@ -183,6 +182,8 @@ def PrepareVMEnvironment(self):
     if self.is_static and self.install_packages:
       self.SnapshotPackages()
     self.BurnCpu()
+    self.RemoteCommand('mkdir -p %s' % vm_util.VM_TMP_DIR)
+    self.SetupPackageManager()
```
Comment (ehankland) [inline perfkitbenchmarker/linux_virtual_machine.py]: Define SetupPackageManager in BaseLinuxMixin?

File: perfkitbenchmarker/linux_virtual_machine.py [hunk 2]
```diff
@@ -738,20 +737,30 @@ class ContainerizedDebianMixin(DebianMixin):
...
+  within a Docker Container.
   Any call to RemoteCommand() will be run within the container
   whereas any call to RemoteHostCommand() will be run in the VM itself.
   """
 
+  def _CheckDockerExists(self):
+    """Returns whether docker is installed or not."""
+    resp, _ = self.RemoteHostCommand('command -v docker', ignore_failure=True,
+                                     suppress_warning=True)
+    if resp[:-1] == "":
```
Comment (ehankland) [inline perfkitbenchmarker/linux_virtual_machine.py]: Maybe strip() instead of [:-1] ?

**Comment by ehankland** [comment]
Couple of small comments, otherwise LGTM

**Comment by gablg1** [comment]
Should be good now @ehankland

**Pull request merged**